Kernel/Devices/PCISerialDevice.cpp

```diff
@@ -26,9 +28,16 @@ void PCISerialDevice::detect()
 
             auto bar_base = PCI::get_BAR(address, board_definition.pci_bar) & ~1;
             // FIXME: We should support more than 1 PCI serial port (per card/multiple devices)
-            s_the = new SerialDevice(IOAddress(bar_base + board_definition.first_offset), 68);
-            if (board_definition.baud_rate != SerialDevice::Baud::Baud38400) // non-default baud
-                s_the->set_baud(board_definition.baud_rate);
+            s_the = new SerialDevice(IOAddress(bar_base + board_definition.first_offset), 68, PCI::get_interrupt_line(address));
+            if (board_definition.baud_rate != SerialDevice::serial_baud_from_termios(TTYDEF_SPEED).value()) {
+                auto maybe_new_baud = SerialDevice::termios_baud_from_serial(board_definition.baud_rate);
+                if (!maybe_new_baud.has_value()) {
+                    dbgln("FIXME: PCISerialDevice's speed is missing from termios");
+                }
+                auto termios = s_the->get_termios();
+                termios.c_ispeed = termios.c_ospeed = maybe_new_baud.value();
```

Comment:
this will ASSERT if maybe_new_baud doesnt have a value, we should probably cancel setting up this serial device instead